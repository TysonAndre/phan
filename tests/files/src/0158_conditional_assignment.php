<?php declare(strict_types=1);

if ( !empty($_SERVER['HOST_NAME']) && $_SERVER['HOST_NAME'] == "www.some.domain" ) {
    $important_variable = true;
}
'@phan-debug-var $important_variable';
$use_variable = $important_variable;
echo count($use_variable);
