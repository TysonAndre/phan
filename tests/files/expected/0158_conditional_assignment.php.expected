%s:3 PhanDebugAnnotation @phan-debug-var requested for variable $important_variable - it has union type ?true=(real=?true=)
%s:7 PhanPossiblyUndeclaredGlobalVariable Global variable $important_variable is possibly undeclared
%s:8 PhanTypeMismatchArgumentInternal Argument 1 ($value) is $use_variable of type ?true but \count() takes \Countable|\ResourceBundle|\SimpleXMLElement|array
